internal/fields/mappings_test.go

```diff
@@ -0,0 +1,594 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package fields
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/elastic/elastic-package/internal/logger"
+)
+
+func TestComparingMappings(t *testing.T) {
+	cases := []struct {
+		title          string
+		preview        map[string]any
+		actual         map[string]any
+		schema         []FieldDefinition
+		expectedErrors []string
+	}{
+		{
+			title: "same mappings",
+			preview: map[string]any{
+				"@timestamp": map[string]any{
+					"type": "keyword",
+				},
+				"host": map[string]any{
+					"properties": map[string]any{
+						"name": map[string]any{
+							"type": "text",
+						},
+					},
+				},
+				"file": map[string]any{
+					"properties": map[string]any{
+						"path": map[string]any{
+							"type": "text",
+						},
+					},
+				},
+				"foo": map[string]any{
+					"type": "keyword",
+					"fields": map[string]any{
+						"text": map[string]any{
+							"type": "match_only_text",
+						},
+					},
+				},
+			},
+			actual: map[string]any{
+				"@timestamp": map[string]any{
+					"type": "keyword",
+				},
+				"host": map[string]any{
+					"properties": map[string]any{
+						"name": map[string]any{
+							"type": "text",
+						},
+					},
+				},
+				"file": map[string]any{
+					"properties": map[string]any{
+						"path": map[string]any{
+							"type": "text",
+						},
+					},
+				},
+				"foo": map[string]any{
+					"type": "keyword",
+					"fields": map[string]any{
+						"text": map[string]any{
+							"type": "match_only_text",
+						},
+					},
+				},
+			},
+			schema:         []FieldDefinition{},
+			expectedErrors: []string{},
+		},
+		{
+			title: "validate field with ECS",
+			preview: map[string]any{
+				"foo": map[string]any{
+					"type": "keyword",
+				},
+			},
+			actual: map[string]any{
```

Comment:
I was thinking that maybe this could happen since in system testing it is appended all the ECS definitions depending on the stack version or `import_mappings`. 

Maybe there could be a case (e.g. with the `ecs@mappings` component template without `import_mappings`) where some field does not match any dynamic template but it is in the ECS.

Would that make sense @jsoriano ?

Reply:
I mean that in this `actual` mapping, the `foo` field should be also present, right?